syntax = "proto3";

package odin.v1;

import "odin/v1/common.proto";
import "odin/v1/options.proto";

// Raw audio data chunks from microphone input
message AudioBufferMic {
  option (odin.v1.nats_subject) = "audio.buffer.mic";
  option (odin.v1.nats_jetstream) = true;

  AudioData audio = 1;
  string client_id = 2;
}

// Generated speech audio from TTS
message AudioBufferSpeaker {
  option (odin.v1.nats_subject) = "audio.buffer.speaker";
  option (odin.v1.nats_jetstream) = true;

  AudioData audio = 1;
  SessionInfo session = 2;
}

// Audio visualization data for frontend
message AudioBufferSession {
  option (odin.v1.nats_subject) = "audio.buffer.session";

  AudioData audio = 1;
  SessionInfo session = 2;
}